### PR TITLE
Feature/migrate to rxjva 3

### PR DESCRIPTION
**Pull request by klamborowski**
Working on migration to rxjva3. Currently WIP and waiting for navi migration: https://github.com/trello/navi/pull/106

Issue #315

### CODE REVIEW

**Comment by klamborowski** [comment]
@dlew alternatively we can just remove rxlifecycle-navi module and drop navi support.

**Comment by dlew** [comment]
Thanks for the PR - just a warning that I'm *very* busy right now and probably won't be able to review it for a bit. I'll put it on my queue, just know I'm not ignoring this.

**Comment by nongdenchet** [comment]
Before merging this to master, should we create a separate branch for `2.x` something similar to `1.x` version https://github.com/trello/RxLifecycle/tree/1.x

File: gradle.properties [hunk 1]
```diff
@@ -11,4 +11,7 @@ POM_LICENCE_URL=http://www.apache.org/licenses/LICENSE-2.0.txt
 POM_LICENCE_DIST=repo
 
 POM_DEVELOPER_ID=dlew
-POM_DEVELOPER_NAME=Dan Lew
\ No newline at end of file
+POM_DEVELOPER_NAME=Dan Lew
+
+android.enableJetifier=true
```
[the remark this answers is not in the document]
Comment (klamborowski) [reply gradle.properties:16]: It's not required, I will remove it in next push.

File: rxlifecycle/src/test/java/com/trello/rxlifecycle3/UntilCorrespondingEventTransformerFlowableTest.java [hunk 2]
```diff
@@ -42,7 +42,6 @@ public void noEvents() {
         stream.onNext("1");
         stream.onNext("2");
         testSubscriber.assertValues("1", "2");
-        testSubscriber.assertNotTerminated();
```
[the remark this answers is not in the document]
Comment (klamborowski) [reply rxlifecycle/src/test/java/com/trello/rxlifecycle3/UntilCorrespondingEventTransformerFlowableTest.java:45]: Done.

File: gradle.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 GROUP=com.trello.rxlifecycle3
-VERSION_NAME=3.1.0-SNAPSHOT
+VERSION_NAME=3.3.0-SNAPSHOT
```
[the remark this answers is not in the document]
Comment (klamborowski) [reply gradle.properties]: Done.

**Pull request merged**